NumLib/ODESolver/TimeDiscretization.h

```diff
@@ -200,6 +211,25 @@ class TimeDiscretization
      */
     virtual bool needsPreload() const { return false; }
     //! @}
+
+protected:
+    std::unique_ptr<GlobalVector> _dx; ///< Used to store \f$ u_{n+1}-u_{n}\f$.
+
+    /**
+      * Compute and return the relative change of solutions between two successive
+      * time steps by \f$ e_n = \|u^{n+1}-u^{n}\|/\|u^{n+1}\| \f$.
+      *
+      * @param x         The current solution
+      * @param x_old     The previous solution
+      * @param norm_type The norm type of global vector
+      * @return          \f$ e_n = \|u^{n+1}-u^{n}\|/\|u^{n+1}\| \f$.
+      *
+      * @warning the value of x_old is changed to x - x_old after this computation.
+    */
+    double computeRelativeChangeFromPreviousTimestep(
```

Comment:
Is it necessary to put this function into the interface; So far I have seen it is only used in the .cpp file, in the implementation.